File: ios/Classes/HandleClickUrlStreamHandler.swift
import Foundation
import Flutter

class HandleClickUrlStreamHandler: NSObject, FlutterStreamHandler{
    var sink: FlutterEventSink?
    
    func onListen(withArguments arguments: Any?, eventSink events: @escaping FlutterEventSink) -> FlutterError? {
        sink = events
        return nil
    }
    
    @objc func onUrlClicked(url: String) {
        guard let sink = sink else { return }
        sink(url)
    }
    
    func onCancel(withArguments arguments: Any?) -> FlutterError? {
        sink = nil
        return nil
    }
}

File: android/src/main/kotlin/com/iadvize/iadvize_flutter_sdk/HandleClickUrlStreamHandler.kt
package com.iadvize.iadvize_flutter_sdk

import android.os.Handler
import io.flutter.plugin.common.EventChannel
import java.util.*

class HandleClickUrlStreamHandler: EventChannel.StreamHandler {
    var sink: EventChannel.EventSink? = null

    fun onUrlClicked(value: String) {
        sink?.success(value)
    }

    override fun onListen(arguments: Any?, events: EventChannel.EventSink?) {
        sink = events
    }

    override fun onCancel(arguments: Any?) {
        sink = null
    }
}